go/oasis-node/cmd/common/metrics/config/config.go

```diff
@@ -0,0 +1,61 @@
+// Package config implements global metrics configuration options.
+package config
+
+import (
+	"fmt"
+	"time"
+)
+
+// Config is the metrics configuration structure.
+type Config struct {
+	// Metrics mode (none, pull, push).
+	Mode string `yaml:"mode"`
+	// Metrics pull address.
+	Address string `yaml:"address"`
+
+	// Metrics push job name (debug-only).
+	JobName string `yaml:"job_name,omitempty"`
+	// Metrics push instance labels (debug-only).
+	Labels map[string]string `yaml:"labels,omitempty"`
+	// Metrics push interval (debug-only).
+	Interval time.Duration `yaml:"interval,omitempty"`
+}
+
+// Validate validates the configuration settings.
+func (c *Config) Validate() error {
+	switch c.Mode {
+	case "none":
+	case "pull":
```

Comment:
Could use constants here, e.g. by moving them to a separate package `constants` to avoid import cycles.